### PR TITLE
Added some React coding guidelines

### PR DESCRIPTION
**Pull request by bartaz**
First basic coding guidelines for React.

### CODE REVIEW

File: style/react.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+Also OK:
+
+```js
+module.exports = React.createClass({
+  displayName: 'MyComponent',
+
+  render: function(){ /* ... */ }
+});
+```
```
Comment (bartaz) [inline style/react.md]: I'm not sure which one I prefer more:

This one avoids using additional property in the class, but on the other hand depends on 'magic' of JSX to guess the name of the class:
```js
const MyComponent = React.createClass({ /*...*/ });
module.exports = MyComponent;
```

This one required additional explicit property with the name, but at least doesn't rely on JSX to guess the name from variable assignment:
```js
module.exports = React.createClass({
  displayName: 'MyComponent',
  /*...*/
});
```

Which one looks better for you @earnubs @squidsoup ?

Comment (earnubs) [inline style/react.md]: I prefer the JSX magic.

**Comment by earnubs** [comment]
+1

File: style/react.md (hunk 1 above)
Comment (bartaz) [inline style/react.md]: Agreed, I'll remove `displayName` example and stick to one preferred style.

Comment (squidsoup) [inline style/react.md]: +1 to not assigning directly to module.exports, as if you forget to explicitly set displayName, you get no feedback in the react devtools. One less thing to remember.

**Pull request merged**